pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -3356,13 +3367,24 @@ public void unblockDispatchersOnUnAckMessages(List<PersistentDispatcherMultipleC
     }
 
     private static class ConfigField {
+        // field holds the pulsar dynamic configuration.
         final Field field;
+
+        // customValue holds the external dynamic configuration.
+        String customValue;
```

Comment:
Should it be volatile?

```suggestion
        volatile String customValue;
```